Feat/google support (#10)
superfaceai/sdk: typescript/src/google/README.md

> [Frameworks](../../../) → **Google GenAi** → **TypeScript** | Python (soon)

Reviewer: ```suggestion
[Frameworks](../../../) → **Google Gemini** → **TypeScript** | Python (soon)
```

> # <img src="https://github.com/superfaceai/sdk/raw/main/docs/logos/google.png" alt="Google GenAi" width="30" height="30" /> Superface SDK for Google GenAi

Reviewer: ```suggestion
# <img src="https://github.com/superfaceai/sdk/raw/main/docs/logos/google.png" alt="Google Gen AI SDK (Gemini)" width="30" height="30" /> Superface SDK for Google Gemini (Gen AI)
```

> Superface SDK lets you integrate external apps with your Google GenAi agent using function calling.

Reviewer: ```suggestion
Superface SDK lets you integrate external apps with your Google Gemini agent using function calling.
```

> This SDK is optimized to be used with Google GenAi SDK or other Google GenAi-compatible LLM SDKs.

Reviewer: ```suggestion
This SDK is optimized to be used with Google Gen AI SDK or other Google Gen AI-compatible LLM SDKs.
```

Although maybe this can be removed entirely as I don't know of anyone who tries mimicking Google's SDK (as opposed to OpenAI's)

## Installation

```sh
npm install superface
```

## Usage

Obtain your agent's secret key and connect tools you want your agent to have access to in [Superface dashboard](https://pod.superface.ai/hub/api).

```ts
import Superface from 'superface/google';

const superface = new Superface({
  apiKey: process.env.SUPERFACE_API_KEY,
});
```

Load connected tools for agent using [`getTools()` method](#gettools).

```ts
const tools = await superface.getTools();
```

> Or use directly in Google GenAi chat completion call

Reviewer: ```suggestion
Or use directly in Google Gen AI chat completion call
```

```ts
import { GoogleGenAI, ContentUnion } from '@google/genai';
const googleGenAi = new GoogleGenAI({ apiKey: GEMINI_API_KEY });

const messages: ContentUnion[] = [
  {
    role: 'user',
    parts: [...]
  }
]

const chatCompletion = await googleGenAi.models.generateContent({
  model: 'gemini-2.0-flash',
  contents: messages,
  config: {
    tools: await superface.getTools()
  }     
});
```

## Handling tool calls

> When using [function calling](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling) with raw Google GenAi SDK, you need to handle your tools calls manually:

Reviewer: ```suggestion
When using [function calling](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling) with raw Google Gen AI SDK, you need to handle your tools calls manually:
```
